### PR TITLE
test: pass `protocol` in `Puppeteer.connect` in tests

### PR DESCRIPTION
**Pull request by sadym-chromium**
Add an internal field `protocol: 'cdp' | 'webDriverBiDi'` to the `Browser` and use is in tests when `Puppeter.connect(...)` is called.

### CODE REVIEW

File: packages/puppeteer-core/src/cdp/Browser.ts [hunk 1]
```diff
@@ -345,8 +347,8 @@ export class CdpBrowser extends BrowserBase {
     return this.#connection.url();
   }
 
-  override async newPage(): Promise<Page> {
-    return await this.#defaultContext.newPage();
+  override newPage(): Promise<Page> {
+    return this.#defaultContext.newPage();
```
Comment (sadym-chromium) [inline packages/puppeteer-core/src/cdp/Browser.ts]: Was required by `eslint`

Comment (Lightning00Blade) [reply packages/puppeteer-core/src/cdp/Browser.ts]: Readd `async` to the function and EsLint will enforce the `await`. Or there is another rule (maybe your IDE adds) that makes this disappear ?

File: packages/puppeteer-core/src/cdp/BrowserConnector.ts [hunk 2]
```diff
@@ -29,6 +29,8 @@ import {isErrorLike} from '../util/ErrorLike.js';
 import {CdpBrowser} from './Browser.js';
 import {Connection} from './Connection.js';
 import type {ConnectOptions} from './ConnectOptions.js';
+import {BidiBrowser} from '../bidi/Browser';
```
Comment (Lightning00Blade) [inline packages/puppeteer-core/src/cdp/BrowserConnector.ts]: We should be doing this part with dynamic import similar to `launch`. Note Type imports are allowed

File: packages/puppeteer-core/src/cdp/Browser.ts (hunk 1 above)
Comment (sadym-chromium) [reply packages/puppeteer-core/src/cdp/Browser.ts]: During `npm run format`,  the await is automatically added, after which another rule is happened to be broken:
>  514:12  error  Returning an awaited value that is not a promise is not allowed  @typescript-eslint/return-await

File: packages/puppeteer-core/src/cdp/BrowserConnector.ts (hunk 2 above)
Comment (sadym-chromium) [reply packages/puppeteer-core/src/cdp/BrowserConnector.ts]: Rolled back that part and focused only on the tests in this PR

File: packages/puppeteer-core/src/cdp/Browser.ts (hunk 1 above)
Comment (sadym-chromium) [reply packages/puppeteer-core/src/cdp/Browser.ts]: never mind, looks like it was a problem with local setup which is now gone

**Review by OrKoN — CHANGES_REQUESTED** [review]
Please also update expectations for these new failures https://github.com/puppeteer/puppeteer/actions/runs/6824148748/job/18559474328?pr=11342

**Review by OrKoN — APPROVED** [review]

**Pull request merged**